### PR TITLE
[test] Minor updates to pytest usage

### PR DESCRIPTION
**Pull request by tjaychen**
- Add logging varibale
- Allow pytest to log to file when variable supplied

### CODE REVIEW

**Comment by tjaychen** [comment]
split out from #877

File: test/systemtest/conftest.py [hunk 1]
```diff
@@ -125,3 +126,8 @@ def spiflash(pytestconfig):
     path = Path(pytestconfig.getoption('spiflash')).resolve()
     assert path.is_file()
     return path
+
+@pytest.fixture(scope="session")
+def logfile(pytestconfig):
+    """Return path to logfile."""
+    return pytestconfig.getoption('log')
```
Comment (mcy) [inline test/systemtest/conftest.py]: All the other fixtures return a Path; I'd suggest doing something like 
```python
log = pytestconfig.getoption('log')
if log == '':
  return
path = Path(log).resolve()
assert not path.is_directory()
return path
```

File: test/systemtest/functional_fpga_test.py [hunk 2]
```diff
@@ -53,13 +53,22 @@ def spiflash_proc(self, tmp_path, spiflash, sw_test_bin):
...
         Executes the binary and inspects its UART for "PASS!\r\n" or "FAIL!\r\n".
         """
 
         logger = logging.getLogger(__name__)
 
+        if logfile:
+            logger.debug("Logfile at" + logfile)
+            # create logger with 'spam_application'
+            logger.setLevel(logging.DEBUG)
+            # create file handler which logs even debug messages
```
Comment (mcy) [inline test/systemtest/functional_fpga_test.py]: Format both these comments as complete sentences (start with capitalized letter, end with period).

File: test/systemtest/functional_fpga_test.py [hunk 3]
```diff
@@ -53,13 +53,22 @@ def spiflash_proc(self, tmp_path, spiflash, sw_test_bin):
...
-    def test_execute_binary(self, spiflash_proc, fpga_uart, uart_timeout):
+    def test_execute_binary(self, spiflash_proc, fpga_uart, uart_timeout, logfile):
         """
         Executes the binary and inspects its UART for "PASS!\r\n" or "FAIL!\r\n".
         """
 
         logger = logging.getLogger(__name__)
 
+        if logfile:
+            logger.debug("Logfile at" + logfile)
```
Comment (mcy) [inline test/systemtest/functional_fpga_test.py]: You're missing a space, write `"Logfile at " + logfile` (or, for extra clarity, `"Logfile at %s" % (logfile,)`, but that one's kind of overkill).

File: test/systemtest/functional_verilator_test.py [hunk 4]
```diff
@@ -48,13 +48,23 @@ def sim_top_earlgrey(self, tmp_path, sim_top_build, sw_test_bin, rom_bin):
 
         p_sim.terminate()
 
-    def test_execute_binary(self, sim_top_earlgrey, uart_timeout):
+    def test_execute_binary(self, sim_top_earlgrey, uart_timeout, logfile):
         """
         Executes the binary and inspects its UART for "PASS!\r\n" or "FAIL!\r\n".
         """
 
+        #if logfile:
```
Comment (mcy) [inline test/systemtest/functional_verilator_test.py]: Looks like you left this in?

File: test/systemtest/functional_fpga_test.py [hunk 5]
```diff
@@ -53,13 +53,22 @@ def spiflash_proc(self, tmp_path, spiflash, sw_test_bin):
...
+            # create logger with 'spam_application'
+            logger.setLevel(logging.DEBUG)
+            # create file handler which logs even debug messages
+            fh = logging.FileHandler(filename=logfile, mode='w')
+            fh.setLevel(logging.DEBUG)
+            logger.addHandler(fh)
```
Comment (mcy) [inline test/systemtest/functional_fpga_test.py]: This block is repeated twice here and in the Verilator fixture. Consider factoring this into a helper function in test_util?

**Comment by imphil** [comment]
pytest does have built-in logging functionality (https://docs.pytest.org/en/latest/logging.html). Have you considered using that?

**Comment by tjaychen** [comment]
ah, since @imphil pointed out pytest already has logging functions, let me look at the first to see if it fits the requirements better.

**Comment by tjaychen** [comment]
actually thinking about it more.. since the existing code is already using a logger, i don't want to go in and change it all out.  Let me update per @mcy's comments first, and then maybe as a separate PR investigate switching everything to pytest native logging.

File: test/systemtest/conftest.py (hunk 1 above)
Comment (tjaychen) [reply test/systemtest/conftest.py]: fixed, thanks for the suggestion.

File: test/systemtest/functional_fpga_test.py (hunk 2 above)
Comment (tjaychen) [reply test/systemtest/functional_fpga_test.py]: fixed

File: test/systemtest/functional_verilator_test.py (hunk 4 above)
Comment (tjaychen) [reply test/systemtest/functional_verilator_test.py]: fixed

File: test/systemtest/functional_fpga_test.py (hunk 5 above)
Comment (tjaychen) [reply test/systemtest/functional_fpga_test.py]: moved

**Pull request merged**